File: support/mock_backend/devLib/scrollPhat.c
#include <stdint.h>
#include <stddef.h>
#include <stdbool.h>
#include <string.h>
#include "devLib/scrollPhat.h"
void scrollPhatPoint(int x, int y, int colour) { (void)x;(void)y;(void)colour; }
void scrollPhatLine(int x0, int y0, int x1, int y1, int colour) { (void)x0;(void)y0;(void)x1;(void)y1;(void)colour; }
void scrollPhatLineTo(int x, int y, int colour) { (void)x;(void)y;(void)colour; }
void scrollPhatRectangle(int x1, int y1, int x2, int y2, int colour, int filled) { (void)x1;(void)y1;(void)x2;(void)y2;(void)colour;(void)filled; }
void scrollPhatUpdate(void) {  }
void scrollPhatClear(void) {  }
int scrollPhatPutchar(int c) { (void)c;return 1; }
void scrollPhatPuts(const char * str) { (void)str; }
void scrollPhatPrintf(const char * message, ...) { (void)message; }
void scrollPhatPrintSpeed(const int cps10) { (void)cps10; }
void scrollPhatIntensity(const int percent) { (void)percent; }
int scrollPhatSetup(void) { return 1; }

File: tests/devLib/scrollPhat.py
import importlib
import pytest

m=importlib.import_module("wiringop.dev_lib.scroll_phat")

def test_scroll_phat_point():
    _=m.scroll_phat_point(1, 1, 1)

def test_scroll_phat_line():
    _=m.scroll_phat_line(1, 1, 1, 1, 1)

def test_scroll_phat_line_to():
    _=m.scroll_phat_line_to(1, 1, 1)

def test_scroll_phat_rectangle():
    _=m.scroll_phat_rectangle(1, 1, 1, 1, 1, 1)

def test_scroll_phat_update():
    _=m.scroll_phat_update()

def test_scroll_phat_clear():
    _=m.scroll_phat_clear()

def test_scroll_phat_putchar():
    _=m.scroll_phat_putchar(1)

def test_scroll_phat_puts():
    _=m.scroll_phat_puts("x")

def test_scroll_phat_printf():
    m.scroll_phat_printf("safe text")

def test_scroll_phat_print_speed():
    _=m.scroll_phat_print_speed(1)

def test_scroll_phat_intensity():
    _=m.scroll_phat_intensity(1)

def test_scroll_phat_setup():
    _=m.scroll_phat_setup()
